core/src/main/scala/org/apache/spark/sql/SnappyParser.scala

```diff
@@ -606,33 +606,33 @@ class SnappyParser(session: SnappySession)
     identifier ~ (
-        '(' ~ ws ~ (
-            '*' ~ ws ~ ')' ~ ws ~> ((udfName: String) =>
-              if (udfName.equalsIgnoreCase("COUNT")) {
-                AggregateExpression(Count(Literal(1, IntegerType)),
-                  mode = Complete, isDistinct = false)
-              } else {
-                throw Utils.analysisException(s"invalid expression $udfName(*)")
-              }) |
-            (DISTINCT ~> trueFn).? ~ (expression * commaSep) ~ ')' ~ ws ~
-                (OVER ~ windowSpec).? ~> { (u: Any, d: Any, e: Any, w: Any) =>
-              val udfName = u.asInstanceOf[String]
-              val exprs = e.asInstanceOf[Seq[Expression]]
-              val function = if (d.asInstanceOf[Option[Boolean]].isEmpty) {
-                UnresolvedFunction(udfName, exprs, isDistinct = false)
-              } else if (udfName.equalsIgnoreCase("COUNT")) {
-                aggregate.Count(exprs).toAggregateExpression(isDistinct = true)
-              } else {
-                UnresolvedFunction(udfName, exprs, isDistinct = true)
-              }
-              w.asInstanceOf[Option[WindowSpec]] match {
-                case None => function
-                case Some(spec: WindowSpecDefinition) =>
-                  WindowExpression(function, spec)
-                case Some(ref: WindowSpecReference) =>
-                  UnresolvedWindowExpression(function, ref)
-              }
+      ('.' ~ identifier).? ~ '(' ~ ws ~ (
+        '*' ~ ws ~ ')' ~ ws ~> ((n1: String, n2: Option[String]) =>
+          if (n1.equalsIgnoreCase("COUNT")) {
+            AggregateExpression(Count(Literal(1, IntegerType)),
+              mode = Complete, isDistinct = false)
+          } else {
+            throw Utils.analysisException(s"invalid expression $n1(*)")
+          }) |
+          (DISTINCT ~> trueFn).? ~ (expression * commaSep) ~ ')' ~ ws ~
+            (OVER ~ windowSpec).? ~> { (n1: String, n2: Option[String], d: Any, e: Any, w: Any) =>
+            val udfName = n2.fold(new FunctionIdentifier(n1))(new FunctionIdentifier(_, Some(n1)))
+            val exprs = e.asInstanceOf[Seq[Expression]]
+            val function = if (d.asInstanceOf[Option[Boolean]].isEmpty) {
+              UnresolvedFunction(udfName, exprs, isDistinct = false)
+            } else if (udfName.funcName.equalsIgnoreCase("COUNT")) {
+              aggregate.Count(exprs).toAggregateExpression(isDistinct = true)
+            } else {
+              UnresolvedFunction(udfName, exprs, isDistinct = true)
             }
+            w.asInstanceOf[Option[WindowSpec]] match {
+              case None => function
+              case Some(spec: WindowSpecDefinition) =>
+                WindowExpression(function, spec)
+              case Some(ref: WindowSpecReference) =>
+                UnresolvedWindowExpression(function, ref)
+            }
+          }
         ) |
         '.' ~ ws ~ (
             identifier. +('.' ~ ws) ~> ((i1: String, rest: Any) =>
```

Comment:
After looking at the parsing looks, change it as follows:

Rename "tableIdentifier" and "functionIdentifier" as "qualifiedIdentifier" and replace all usages of the two by it.

Change "idenfier" in the rule above to "qualifiedIdentifier" and then the parsing rule will look like:

```
qualifiedIdentifier ~ (
  '(' ~ ws ~ (
    ...
  ) |
  '.' ~ ws '*' ~> UnresolvedStar(...) |
  MATCH ~> UnresolvedAttribute.quoted _
...
)
```

i.e. like before just that top-level "identifier" becomes "qualifiedIdentifier" and the additional rule of UnresolvedAttribute gets subsumed in the MATCH case.